commands/system_test.go

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package commands
+
+import (
+	"strconv"
+	"time"
+
+	"github.com/mattermost/mattermost-server/v5/model"
+
+	"github.com/mattermost/mmctl/printer"
+
+	"github.com/spf13/cobra"
+)
+
+func (s *MmctlUnitTestSuite) TestGetBusyCmd() {
+	s.Run("GetBusy when not set", func() {
+		printer.Clean()
+		sbs := &model.ServerBusyState{}
+
+		s.client.
+			EXPECT().
+			GetServerBusy().
+			Return(sbs, &model.Response{Error: nil}).
+			Times(1)
+
+		err := getBusyCmdF(s.client, &cobra.Command{}, []string{})
+		s.Require().Nil(err)
+		s.Require().Len(printer.GetLines(), 1)
+		s.Require().Equal(printer.GetLines()[0], sbs)
+		s.Require().Len(printer.GetErrorLines(), 0)
+	})
+
+	s.Run("GetBusy when set", func() {
+		printer.Clean()
+		const minutes = 15
+		expires := time.Now().Add(time.Minute * minutes).Unix()
+		sbs := &model.ServerBusyState{Busy: true, Expires: expires}
+
+		s.client.
+			EXPECT().
+			GetServerBusy().
+			Return(sbs, &model.Response{Error: nil}).
+			Times(1)
+
+		err := getBusyCmdF(s.client, &cobra.Command{}, []string{})
+		s.Require().Nil(err)
+		s.Require().Len(printer.GetLines(), 1)
+		s.Require().Equal(printer.GetLines()[0], sbs)
+		s.Require().Len(printer.GetErrorLines(), 0)
+	})
+
+	s.Run("GetBusy with error", func() {
+		printer.Clean()
+		s.client.
+			EXPECT().
+			GetServerBusy().
+			Return(nil, &model.Response{Error: &model.AppError{Id: "Mock Error"}}).
+			Times(1)
+
+		err := getBusyCmdF(s.client, &cobra.Command{}, []string{})
+		s.Require().NotNil(err)
+		s.Require().Len(printer.GetLines(), 0)
+		s.Require().Len(printer.GetErrorLines(), 1)
```

Comment:
Maybe I am nitpicking, but wouldn't it be nice to make sure the error you are receiving is the error you are expecting? Not sure if this will make the test too brittle, but may make sense.

Reply:
I did implement this, per our sidebar conversation. I ended up discarding it because it was brittle was what we gain.  Some cases are simple, some not so much.  I did however fix up the cases where I was returning an error and printing an error.